spec/controllers/compound_metric_configurations_controller_spec.rb

```diff
@@ -4,13 +4,31 @@
   let(:reading_group) { FactoryGirl.build(:reading_group, :with_id) }
   let(:kalibro_configuration) { FactoryGirl.build(:kalibro_configuration, :with_id) }
 
+  describe 'allowed_metric_configurations' do
+    let(:metric_configurations) { [
+      FactoryGirl.build(:metric_configuration, reading_group_id: reading_group.id),
+      FactoryGirl.build(:compound_metric_configuration, reading_group_id: reading_group.id),
+      FactoryGirl.build(:hotspot_metric_configuration),
+    ] }
+    let(:allowed_metric_configurations) { metric_configurations[0..1] }
+
+    before :each do
+      MetricConfiguration.expects(:metric_configurations_of).with(kalibro_configuration.id).returns(metric_configurations)
+    end
+
+    it 'is expected to filter out hotspot metric configurations' do
+      expect(subject.send(:allowed_metric_configurations, kalibro_configuration.id)).to eq(allowed_metric_configurations)
+    end
+  end
+
   describe 'new' do
```

Comment:
If you unit tested it, mock it elsewhere. If not, modify tests accordingly so they fail if `allowed_metric_configurations` side effects are not found among the produced ones.

(This is a case where TDD would have been a lot easier).